src/FrontDoor/FrontDoor/Cmdlets/GetAzureRmFrontDoorFrontendEndpoint.cs

```diff
@@ -0,0 +1,90 @@
+﻿using System;
+using System.Collections;
+using System.Management.Automation;
+using System.Net;
+using Microsoft.Azure.Commands.FrontDoor.Common;
+using Microsoft.Azure.Commands.FrontDoor.Helpers;
+using Microsoft.Azure.Commands.FrontDoor.Models;
+using Microsoft.Azure.Commands.FrontDoor.Properties;
+using Microsoft.Azure.Management.FrontDoor;
+using System.Linq;
+using Microsoft.Azure.Commands.ResourceManager.Common.ArgumentCompleters;
+using Microsoft.Azure.Management.Internal.Resources.Utilities.Models;
+
+namespace Microsoft.Azure.Commands.FrontDoor.Cmdlets
+{
+    /// <summary>
+    /// Defines the Get-FrontDoorFrontendEndpoint cmdlet.
+    /// </summary>
+    [Cmdlet("Get", ResourceManager.Common.AzureRMConstants.AzureRMPrefix + "FrontDoorFrontendEndpoint", DefaultParameterSetName = FieldsParameterSet), OutputType(typeof(PSFrontendEndpoint))]
+    public class GetAzureRmFrontDoorFrontendEndpoint : AzureFrontDoorCmdletBase
+    {
+        /// <summary>
+        /// The resource group name of the Front Door.
+        /// </summary>
+        [Parameter(Mandatory = true, ParameterSetName = FieldsParameterSet, HelpMessage = "The resource group name.")]
+        [ValidateNotNullOrEmpty]
+        [ResourceGroupCompleter]
+        public string ResourceGroupName { get; set; }
+
+        /// <summary>
+        /// The Front Door name.
+        /// </summary>
+        [Parameter(Mandatory = true, ParameterSetName = FieldsParameterSet, HelpMessage = "Front Door name.")]
+        [ValidateNotNullOrEmpty]
+        [ResourceNameCompleter("Microsoft.Network/frontdoors", nameof(ResourceGroupName))]
+        public string FrontDoorName { get; set; }
+
+        /// <summary>
+        /// Gets or sets the frontend endpoint name.
+        /// </summary>
+        [Parameter(Mandatory = false, HelpMessage = "Frontend endpoint name.")]
+        [ValidateNotNullOrEmpty]
+        public string Name { get; set; }
+
+        /// <summary>
+        ///The Front Door object.
+        /// </summary>
+        [Parameter(Mandatory = true, ParameterSetName = ObjectParameterSet, ValueFromPipeline = true, HelpMessage = "The FrontDoor object.")]
+        [ValidateNotNullOrEmpty]
+        public PSFrontDoor FrontDoorObject { get; set; }
+
+        /// <summary>
+        /// Resource Id of the Front Door
+        /// </summary>
+        [Parameter(Mandatory = true, ParameterSetName = ResourceIdParameterSet, ValueFromPipelineByPropertyName = true, HelpMessage = "Resource Id of the Front Door")]
```

Comment:
@pichandwork this should be the resource id of the frontend endpoint, not the front door. This also shouldn't be in the same parameter set as `-Name`.